packages/legend-query-builder/src/stores/QueryBuilderResultState.ts

```diff
@@ -365,4 +400,23 @@ export class QueryBuilderResultState {
       this.isGeneratingPlan = false;
     }
   }
+
+  *getSql(): GeneratorFn<void> {
+    try {
+      if (!this.executionPlanState.plan || this.checkForStaleResults) {
```

Comment:
This is because the user might have already clicked 'generate plan' but if the query changes, the generated plan will still exist so when the user clicks 'show sql' the SQL shown will be stale and belong to the old sql of the stale query, so it might have potentially changed (similar to how the results panel will show 'results may potentially be stale' upon query change :)